### PR TITLE
Replace go-swagger with swagger-gen

### PR DESCRIPTION
**Pull request by dnephin**
This PR replaces the tool used for `api/types` code generate. We have been using `go-swagger` but this has not worked well:
* the generation includes an iteration over a map. When the order changes the build will fail when it tries to validate the vendoring of the generated code was done correctly.
* the above bug can't really be fixed because `go-swagger` removed support for generating only types.
* `go-swagger` used templates templates to generate code, which proved to be difficult to work with, and it never gave us full control over the templates.
* the `go-swagger` defaults required us to add a lot of `x-nullable: false` all over the spec and often ignored `Title` when it was set.

The new tool is currently at https://github.com/dnephin/swagger-gen-types, but I would like to move it to the moby organization on github, since it's primarily only going to be used with moby/moby.

The rules for generating types are [simple](https://github.com/dnephin/swagger-gen-types/blob/master/README.md), and the code is only ~300 lines. It uses the https://github.com/go-openapi for parsing the swagger spec, and https://godoc.org/github.com/dave/jennifer/jen for code generation.

This new tool should make it a lot easier to convert more types to code generation, and makes it a lot easier to customize the code that it generated.

### CODE REVIEW

**Comment by dnephin** [comment]
It looks like yamllint was removed as part of the multi-stage Dockerfile PR, but validation never run because our validation checks for changes before running. 

I'll look at restoring yamllint.

File: hack/validate/gometalinter.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     ".*\\.pb\\.go",
     "dockerversion/version_autogen.go",
     "api/types/container/container_.*",
+    "api/types/.*-generated\.go",
```
Comment (thaJeztah) [inline hack/validate/gometalinter.json]: Should this have double backslashes?

File: api/types/volumes-generated.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// The level at which the volume exists. Either `global` for cluster-wide, or `local` for machine level.
+	Scope string `json:"Scope"`
+	/*
+	   Low-level details about the volume, provided by the volume driver.
+	   Details are returned as a map with key/value pairs:
+	   `{"key":"value","key2":"value2"}`.
+
+	   The `Status` field is optional, and is omitted if the volume driver
+	   does not support this feature.
+	*/
```
Comment (thaJeztah) [inline api/types/volumes-generated.go]: Not a big fan of the block-style comments; seems they're generated by this https://github.com/dave/jennifer/blob/master/jen/comments.go#L85-L93, so not sure there's an easy workaround (other than splitting on newlines and appending a single-line comment for each line)

File: api/types/container/responses-generated.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	Titles []string `json:"Titles,omitempty"`
+}
+
+// OK response to ContainerUpdate operation
+type ContainerUpdateResponse struct {
+	Warnings []string `json:"Warnings,omitempty"`
+}
+
+// OK response to ContainerWait operation
+type ContainerWaitResponse struct {
```
Comment (thaJeztah) [inline api/types/container/responses-generated.go]: Wondering what determines if this is named `ContainerWaitOKResponse` vs `ContainerWaitResponse`.

Although I like "response"; "body" may be more factual (i.e. it's not the full response); do you know where the code is to generate the names?

Basically; should we try to make the names match the old names?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks! Left some thoughts 👍

File: api/types/container/responses-generated.go (hunk 3 above)
Comment (thaJeztah) [reply api/types/container/responses-generated.go]: Comment doesn't look to be correct godoc as well

File: api/types/volumes-generated.go (hunk 2 above)
Comment (dave) [reply api/types/volumes-generated.go]: Yeah if there's a newline in a comment, jennifer will use multi-line syntax. However, if you prefix your comment with `//` it'll be rendered as-is. You'd need to manually add the `//` to the start of every line though. Feel free to ping me if you'd like any more jennifer help!

**Comment by olljanat** [comment]
@dnephin this one needs to be rebased

**Comment by olljanat** [comment]
Derek add label: status/failing-ci

**Comment by cpuguy83** [comment]
This hasn't been touched in a long time and no one seems to be taking it up, so I'm going to go ahead and close this.

**Pull request closed without merging**